src/test.python.bridge.cpp

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+
+#include <memory>
+
+#include <Python.h>
+
+#include <openassetio/log/LoggerInterface.hpp>
+#include <openassetio/hostApi/ManagerImplementationFactoryInterface.hpp>
+#include <openassetio/python/hostApi.hpp>
+
+namespace {
+struct DummyLoggerInterface : openassetio::log::LoggerInterface {
+  void log(Severity severity, const openassetio::Str& message) override {}
+};
+}  // namespace
+
+int main() {
+  Py_Initialize();
```

Comment:
I've just checked what I've done with embedding Python interpreters before, and I've always set Py_SetProgramName before Py_Initialize.

Also called `Py_SetPath` _after_, but this was a GUI application, so wasn't appropriate to get that from an environment variable like in here. But just saying it for completeness.

Reply:
I had a quick play with `Py_SetProgramName` in Windows and Linux.  

Firstly, its 4 extra lines of code (`wchar_t` conversion) - good 'ole C.

In Windows I didn't see any real change in behaviour.

In Linux, on the other hand, if I point `Py_SetProgramName` to the absolute path of the Python interpreter in the Conan package (i.e. the same package that provides the linked `libpython`), then I can remove the `PYTHONHOME` environment variable, since `Py_Initialize` computes paths based on the program name.

On `Py_SetPath` [the docs](https://docs.python.org/3.7/c-api/init.html#c.Py_SetPath) say

>  This is useful if Python is embedded by an application that has full knowledge of the location of all modules

So setting these in the source code would allow us to remove the `PYTHONHOME`  and  `PYTHONPATH` environment variables, but at the cost of passing their values along as compile definitions instead, essentially hardcoding them in the executable.

I don't think adding either of these makes the executable any more "correct", yet would add a bit more complexity to something that already works as is, so I'm of a mind to leave it be, if that's OK?

Reply:
Yeah that's fine if it doesn't benefit you, I just wanted to mention those other parts that I've needed to use before, so good you've had a look over them. Thanks!